src/sentry/static/sentry/app/views/projectFilters/projectFilters.jsx

```diff
@@ -0,0 +1,404 @@
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import moment from 'moment';
+
+import ApiMixin from '../../mixins/apiMixin';
+import IndicatorStore from '../../stores/indicatorStore';
+import GroupTombstones from '../../components/groupTombstones';
+import LoadingError from '../../components/loadingError';
+import LoadingIndicator from '../../components/loadingIndicator';
+import ProjectState from '../../mixins/projectState';
+import StackedBarChart from '../../components/stackedBarChart';
+import {t} from '../../locale';
+import {intcomma} from '../../utils';
+
+import FilterRow from './filterRow';
+import LegacyBrowserFilterRow from './legacyBrowserFilterRow';
+import ProjectFiltersSettingsForm from './projectFiltersSettingsForm';
+
+const ProjectFilters = React.createClass({
+  mixins: [ApiMixin, ProjectState],
+
+  getInitialState() {
+    let until = Math.floor(new Date().getTime() / 1000);
+    let since = until - 3600 * 24 * 30;
+
+    return {
+      expected: 3,
+      loading: true,
+      error: false,
+      statsError: false,
+      filterList: [],
+      querySince: since,
+      queryUntil: until,
+      rawStatsData: null,
+      formattedData: [],
+      projectOptions: {},
+      blankStats: true,
+      activeSection: 'data-filters',
+      tombstones: [],
+      tombstoneError: false
+    };
+  },
+
+  componentWillMount() {
+    this.fetchData();
+  },
+
+  componentDidUpdate(prevProps) {
+    if (!this.state.loading && !this.state.formattedData) {
+      this.render();
```

Comment:
Hmm this file was renamed but git(hub?) doesn't do a good job with rename + changes, sorry if any of these were not related to your changes, these will be general comments and I would rather release this than do big refactors (use your judgement I guess?)

Why are we doing this? seems like we're double rendering?

Reply:
Yeah...should have mentioned that we split up the code and the meat of the changes are actually just in the `formatData()` method. The other components (and most of this one) already existed as is, but we broke them out into separate files since the `projectFilters` view was pretty unwieldy.

I'll look into why this is double rendering, though.

Reply:
Yeah I'm not sure what I was trying to do with this. I'll remove it.